### PR TITLE
Refactoring on GetEventBounds in Widget

### PR DESCRIPTION
**Pull request by teinarss**
Just a small change that i had in a bigger upcoming PR.

### CODE REVIEW

File: OpenRA.Game/Widgets/Widget.cs [hunk 1]
```diff
@@ -279,21 +279,22 @@ public void PostInit(WidgetArgs args)
 
 		public virtual Rectangle EventBounds { get { return RenderBounds; } }
 
-		public virtual Rectangle GetEventBounds()
+		public virtual bool CheckEventBounds(int2 location)
```
Comment (pchote) [inline OpenRA.Game/Widgets/Widget.cs]: `CheckEventBounds` is ambiguous. Can we call this something like `EventBoundsContains`?

**Comment by teinarss** [comment]
Updated

File: OpenRA.Game/Widgets/Widget.cs [hunk 2]
```diff
@@ -279,21 +279,22 @@ public void PostInit(WidgetArgs args)
...
-		public virtual Rectangle GetEventBounds()
+		public virtual bool EventBoundsContains(int2 location)
 		{
 			// PERF: Avoid LINQ.
-			var bounds = EventBounds;
+			if (EventBounds.Contains(location))
+				return true;
+
 			foreach (var child in Children)
 			{
```
Comment (pchote) [inline OpenRA.Game/Widgets/Widget.cs]: Style nit: these braces here and on `child.IsVisible()` can now be removed.

**Review by pchote — DISMISSED** [review]
This will change the behaviour when a child's EventBounds extends outside the parent, but the new behaviour is arguably more sensible than the old.

LGTM, just one minor style nit:

**Comment by teinarss** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Pull request merged**